src/app/Dashboard/Diagnostics/DiagnosticsCard.tsx

```diff
@@ -0,0 +1,204 @@
+    const a = props.actions || [];
+    return [GCButton, ...a];
+  }, [props.actions, GCButton]);
+
+  const gcButtonLoadingProps = React.useMemo(
+    () =>
+      ({
+        spinnerAriaValueText: 'Invoke GC',
+        spinnerAriaLabel: 'saving-credentials',
+        isLoading: running,
+      }) as LoadingProps,
+    [running],
+  );
+
+  const header = React.useMemo(() => {
+    return (
+      <CardHeader
+        actions={{
+          actions: <>{actions}</>,
+          hasNoOffset: false,
+          className: undefined,
+        }}
+      >
+        <CardTitle>
+          {t('DiagnosticsCard.DIAGNOSTICS_CARD_TITLE', {
+            chartKind: props.chartKind,
+            duration: props.duration,
+            period: props.period,
+          })}
+        </CardTitle>
+      </CardHeader>
+    );
+  }, [props.chartKind, props.duration, props.period, t, actions]);
+
+  return isError ? (
+    <ErrorView title={'Error executing diagnostic command'} message={`${errorMessage}`} />
+  ) : (
+    <DashboardCard
+      id={props.chartKind + '-chart-card'}
+      dashboardId={props.dashboardId}
+      cardSizes={DiagnosticsCardSizes}
+      isCompact
+      cardHeader={header}
+      title={props.chartKind}
+      isDraggable={props.isDraggable}
+      isResizable={props.isResizable}
+      isFullHeight={props.isFullHeight}
+    >
+      <CardBody>
+        <Bullseye>
+          <EmptyState variant={EmptyStateVariant.lg}>
+            <EmptyStateHeader
+              titleText={<>{t('DiagnosticsCard.DIAGNOSTICS_CARD_TITLE')}</>}
+              icon={<EmptyStateIcon icon={DataSourceIcon} />}
+              headingLevel="h2"
+            />
+            <EmptyStateBody>
+              <Trans t={t} components={{ label: <Label color="blue" isCompact /> }}>
+                DiagnosticsCard.DIAGNOSTICS_CARD_DESCRIPTION
+              </Trans>
+            </EmptyStateBody>
+            <EmptyStateFooter>
+              <Button variant="primary" onClick={handleGC} {...gcButtonLoadingProps}>
+                {t('DiagnosticsCard.DIAGNOSTICS_GC_BUTTON')}
+              </Button>
+            </EmptyStateFooter>
+          </EmptyState>
+        </Bullseye>
+      </CardBody>
+    </DashboardCard>
+  );
+};
+
+DiagnosticsCard.cardComponentName = 'DiagnosticsCard';
+
+export const DiagnosticsCardSizes: DashboardCardSizes = {
+  span: {
+    minimum: 3,
+    default: 4,
+    maximum: 12,
+  },
+  height: {
+    // TODO: implement height resizing
+    minimum: Number.NaN,
+    default: Number.NaN,
+    maximum: Number.NaN,
+  },
+};
+
+export const DiagnosticsCardDescriptor: DashboardCardDescriptor = {
+  featureLevel: FeatureLevel.BETA,
+  title: 'DiagnosticsCard.DIAGNOSTICS_CARD_TITLE',
+  cardSizes: DiagnosticsCardSizes,
+  description: 'DiagnosticsCard.DIAGNOSTICS_CARD_DESCRIPTION',
+  descriptionFull: 'DiagnosticsCard.DIAGNOSTICS_CARD_DESCRIPTION_FULL',
+  component: DiagnosticsCard,
+  propControls: [],
+  icon: <TachometerAltIcon />,
```

Comment:
I think we already use this Tachometer icon for the JFR metrics cards:

![image](https://github.com/user-attachments/assets/4a73c66d-2e6a-4e43-8984-575ff6bf57d7)

it would probably be better to choose a unique icon.

Maybe the Wrench icon is a good visual for diagnostic actions?

![image](https://github.com/user-attachments/assets/12a53440-1d29-4b7c-9dbe-235ebc09b5ec)

https://www.patternfly.org/design-foundations/icons